File: src/main/java/us/marek/pdf/inputformat/PdfInputFormat.java
package us.marek.pdf.inputformat;

import java.io.IOException;

import org.apache.hadoop.io.LongWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.InputSplit;
import org.apache.hadoop.mapreduce.RecordReader;
import org.apache.hadoop.mapreduce.TaskAttemptContext;
import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;


public class PdfInputFormat extends FileInputFormat<LongWritable, Text> {

    @Override
    public RecordReader<LongWritable, Text> createRecordReader(
            final InputSplit split, final TaskAttemptContext context) throws IOException,
            InterruptedException {

        return new PdfRecordReader();
    }

}

File: src/main/scala/us/marek/pdf/inputformat/PdfInputFormat.scala
package us.marek.pdf.inputformat

import java.io.IOException
import org.apache.hadoop.io.{ LongWritable, Text }
import org.apache.hadoop.mapreduce.{ InputSplit, RecordReader, TaskAttemptContext }
import org.apache.hadoop.mapreduce.lib.input.FileInputFormat

class PdfInputFormat extends FileInputFormat[LongWritable, TikaParsedPdfWritable] {

  // Hadoop Java-land, please excuse annotating checked exceptions :(
  @throws[IOException]
  @throws[InterruptedException]
  def createRecordReader(split: InputSplit, context: TaskAttemptContext): RecordReader[LongWritable, TikaParsedPdfWritable] =
    new PdfRecordReader

}